app/src/main/java/org/mozilla/fenix/onboarding/view/NotificationPermissionDialogScreen.kt

```diff
@@ -0,0 +1,205 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.onboarding.view
+
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material.Icon
+import androidx.compose.material.IconButton
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.google.accompanist.insets.navigationBarsPadding
+import com.google.accompanist.insets.statusBarsPadding
+import org.mozilla.fenix.R
+import org.mozilla.fenix.compose.button.PrimaryButton
+import org.mozilla.fenix.compose.button.SecondaryButton
+import org.mozilla.fenix.theme.FirefoxTheme
+
+/**
+ * Model containing data for the [NotificationPermissionPage].
+ *
+ * @param image [DrawableRes] displayed on the page.
+ * @param title [StringRes] of the permission headline text.
+ * @param description [StringRes] of the permission body text.
+ * @param primaryButtonText [StringRes] of the primary button text.
+ * @param secondaryButtonText [StringRes] of the secondary button text.
+ * @param onRecordImpressionEvent Callback for recording impression event.
+ */
+private data class NotificationPermissionPageState(
+    @DrawableRes val image: Int,
+    @StringRes val title: Int,
+    @StringRes val description: Int,
+    @StringRes val primaryButtonText: Int,
+    @StringRes val secondaryButtonText: Int? = null,
+    val onRecordImpressionEvent: () -> Unit,
+)
+
+/**
+ * A screen for displaying notification pre permission prompt.
+ *
+ * @param onDismiss Invoked when the user clicks on the close or "No thanks" button.
```

Comment:
nit: I wonder if "No thanks" we call it secondary button - a reference like that could quite easily be missed if the text is ever updated